openstef/feature_engineering/cyclic_features.py

```diff
@@ -21,10 +23,68 @@
 )
 logger = structlog.get_logger(__name__)
 
-
+TERRESTRIAL_RADIATION_CSV_PATH: str = (
+    PROJECT_ROOT / "openstef" / "data" / "NL_terrestrial_radiation.csv"
+)
 NUM_SECONDS_IN_A_DAY = 24 * 60 * 60
 
 
+def add_daylight_terrestrial_feature(
+    data: pd.DataFrame,
+    path_to_terrestrial_radiation_csv: str = TERRESTRIAL_RADIATION_CSV_PATH,
+) -> pd.DataFrame:
+    """Add daylight terrestrial radiation feature to the input dataset. This function processes terrestrial radiation
+    data and aligns it with the time indices of the input dataset. The terrestrial radiation data is normalized,
+    interpolated, and merged with the main dataset to provide a feature representing terrestrial radiation.
+
+    Args:
+        data (pd.DataFrame):
+            The input dataset containing a time-indexed DataFrame.
+        path_to_terrestrial_radiation_csv (str):
+            File path to the CSV file containing terrestrial radiation data. The CSV file
+            should have a time-based index.
+
+    Returns:
+        pd.DataFrame:
+            The input dataset with an added column for the terrestrial radiation feature.
+
+    Notes:
+        - The function assumes the input data and the terrestrial radiation data share
+          the same time zone and frequency alignment.
+        - The terrestrial radiation values are normalized using z-score normalization.
+
+    """
+    # Load the terrestrial radiation data
+    terrestrial_radiation = pd.read_csv(path_to_terrestrial_radiation_csv, index_col=0)
+    terrestrial_radiation.index = pd.to_datetime(terrestrial_radiation.index)
+
+    # Create a shifted version of the terrestrial radiation data to align with the input data's year
+    terrestrial_radiation_shifted = terrestrial_radiation.copy()
+    year_diff = data.index.year[0] - terrestrial_radiation.index.year[0]
+    terrestrial_radiation_shifted.index += pd.DateOffset(years=year_diff)
+
+    # Combine original and shifted data, resample to 15-minute intervals, and interpolate missing values
+    terrestrial_radiation = (
+        pd.concat([terrestrial_radiation, terrestrial_radiation_shifted])
+        .resample("15min")
+        .mean()
+        .interpolate()
+    )
```

Comment:
At first, we applied the shifting as described. However, I believe that for the data balancing step, part of the original data might be lost or misaligned when combining it with the shifted data. To avoid this, I think the safest approach is to handle the original and shifted datasets separately.

Reply:
Very good point, there might me a gap of 1 year in the input data (data balancing), we should probably test for that as well. The min range in this case I think would be covered by shift to zero'th index year, anything afterwards needs to be handled adaptively as well.

Reply:
New change fixed it. Thanks